File: csharp/Algorithms/Other/SieveOfEratosthenes.cs
using System;
using System.Collections.Generic;

namespace Algorithms.Other;

public class SieveOfEratosthenes
{
    private readonly bool[] primes;

    public SieveOfEratosthenes(long maximumNumberToCheck)
    {
        primes = new bool[maximumNumberToCheck + 1];

        Array.Fill(this.primes, true, 2, primes.Length - 2);

        for(long i = 2; i * i <= maximumNumberToCheck; i++)
        {
            if (!primes[i])
            {
                continue;
            }

            for(long composite = i * i; composite <= maximumNumberToCheck; composite += i)
            {
                primes[composite] = false;
            }
        }
    }

    public long MaximumNumber => primes.Length - 1;

    public bool IsPrime(long numberToCheck) => primes[numberToCheck];

    public IEnumerable<long> GetPrimes()
    {
        for(long i = 2; i < primes.Length; i++)
        {
            if (primes[i])
            {
                yield return i;
            }
        }
    }
}

File: javascript/Dynamic-Programming/SieveOfEratosthenes.js
function sieveOfEratosthenes(n) {
  if (n <= 1) return []
  const primes = new Array(n + 1).fill(true) 
  primes[0] = primes[1] = false 
  for (let i = 2; i * i <= n; i++) {
    if (primes[i]) {
      for (let j = i * i; j <= n; j += i) {
        primes[j] = false
      }
    }
  }

  return primes.reduce((result, isPrime, index) => {
    if (isPrime) {
      result.push(index)
    }
    return result
  }, [])
}

export { sieveOfEratosthenes }
